tests/filecheck/backend/convert_riscv_scf_to_riscv_cf.mlir

```diff
@@ -0,0 +1,84 @@
+// RUN: xdsl-opt -p convert-riscv-scf-to-riscv-cf --split-input-file %s | filecheck %s
+
+// sum(range(arg0, arg1))
+
+builtin.module {
```

Comment:
This is great, but I'd argue the simplest example to see the lowering (especially the labels, etc) would be to reduce the "noise" from loop-carried args and other statements.

Could you add something like this, please?

```mlir
riscv_func.func @foo(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>) {
  %2 = riscv.li 1 : () -> !riscv.reg<a2>
  %3 = riscv.li 0 : () -> !riscv.reg<a3>
  riscv_scf.for %4 : !riscv.reg<a4> = %0 to %1 step %2 {
      %5 = riscv.add %2, %3 : (!riscv.reg<a2>, !riscv.reg<a3>) -> !riscv.reg<a3>
      riscv_scf.yield
  }
  riscv_func.return
}
```

You can use this as is, I've tried it.

Reply:
I agree that it would be good to have a loop with no iter args, I'll add one.